### DISCUSSION TITLE
Possible to plot data already in DuckDB server?

### DISCUSSION THREAD

**Discussion by mhkeller** [uwdata/mosaic#205] [Q&A] [answered]
I'm looking at creating a workflow where another process creates a table in DuckDB and then I would visualize that table with Mosaic. Essentially, I want to skip the part in the examples where it loads data from a csv or parquet file and subsequently creates a DuckDB table. I'll be using the same DuckDB server setup so that Mosaic still has the same API to the database.

I'm wondering if there are any caveats to this approach. Importantly, the table needs to be added to the coordinator.

Looking at the source for [`create`](https://github.com/uwdata/mosaic/blob/819f69f6e652b7b7846618e65a7635e826427a9a/packages/sql/src/load/create.js), this function constructs a string that is something like:

```sql
CREATE TEMP TABLE IF NOT EXISTS penguins AS SELECT * FROM read_csv('https://raw.githubusercontent.com/uwdata/mosaic/main/data/penguins.csv', auto_detect=true, sample_size=-1)
```
 
Assuming I have already loaded penguins earlier, I'm guessing I could do something like this:

```js
vg.coordinator().exec('CREATE TABLE IF NOT EXISTS penguins AS SELECT * FROM penguins');
```

Purely from DuckDB's standpoint, it doesn't complain about a `CREATE` statement like that but it does feel a little hacky and I'm wondering if there are some nuances that Mosaic will not like. Is this an okay approach or am I missing something?

**Comment by jheer** [comment, marked as answer]
The coordinator doesn't need to "know" about the table ahead of time -- `exec` queries simply pass right through to DuckDB, and there's no need to create a table that already exists. You can simply create tables however you like, then go straight to visualizing them!

**Reply by mhkeller** [reply]
Thanks! I’ll try wiring it up.
